test/vtgatev2_test.py

```diff
@@ -1050,34 +1148,47 @@ def test_restart_mysql_tablet_idle(self):
     # then restart replication, and write data, make sure we go back to healthy
     for t in [self.replica_tablet]:
       utils.run_vtctl(['StartSlave', t.tablet_alias])
-      utils.wait_for_tablet_type(t.tablet_alias, 'replica')
+      utils.run_vtctl(['RunHealthCheck', t.tablet_alias, 'replica'],
+                      auto_log=True)
       t.wait_for_vttablet_state('SERVING')
+    if vtgate_gateway_flavor().flavor() != 'shardgateway':
+      wait_for_vars(
+          vtgate_gateway_flavor().connection_count_vars(),
+          '%s.%s.replica' % (KEYSPACE_NAME, SHARD_NAMES[self.shard_index]),
+          1)
     vtgate_conn._execute(
         'select 1 from vt_insert_test', {},
         KEYSPACE_NAME, 'replica',
         keyranges=[self.keyrange])
-    self.replica_tablet.kill_vttablet()
-    self.tablet_start(self.replica_tablet, 'replica')
-    self.replica_tablet.wait_for_vttablet_state('SERVING')
-    # TODO(liguo): expect to fail
-    # until we can detect vttablet shuts down gracefully
-    # while VTGate is idle.
-    # NOTE: with grpc, it will reconnect, and not trigger an error.
-    if protocols_flavor().tabletconn_protocol() == 'grpc':
-      return
-    try:
-      result = vtgate_conn._execute(
-          'select 1 from vt_insert_test', {},
-          KEYSPACE_NAME, 'replica',
-          keyranges=[self.keyrange])
-      self.fail(
-          'DatabaseError should have been raised, but got %s' % str(result))
-    except Exception, e:  # pylint: disable=broad-except
-      self.assertIsInstance(e, dbexceptions.DatabaseError)
-      self.assertNotIsInstance(e, dbexceptions.IntegrityError)
-      self.assertNotIsInstance(e, dbexceptions.OperationalError)
-      self.assertNotIsInstance(e, dbexceptions.TimeoutError)
+
+    if (protocols_flavor().tabletconn_protocol() != 'grpc'
+        and vtgate_gateway_flavor().flavor() == 'shardgateway'):
```

Comment:
Can you please document this restriction? The gRPC one is mentioned below, the shardgateway one might be surprising.